rfd/xxxx-automatic-azure-server-discovery.md

```diff
@@ -0,0 +1,105 @@
+---
+authors: Andrew Burke (andrew.burke@goteleport.com)
+state: draft
+---
+
+# RFD X - Automatic discovery of Azure servers
+
+## What
+
+Teleport SSH services will be able to automatically discover Azure virtual machine
+instances. See [RFD 57](https://github.com/gravitational/teleport/blob/master/rfd/0057-automatic-aws-server-discovery.md)
+(Automatic discovery and enrollment of AWS servers) for the same feature implemented
+for AWS.
+
+Note: unlike RFD 57, this RFD covers discovery only. Azure node joining will be added
+later.
+
+## Why
+
+RFD 57 replaced the process of manually installing Teleport on EC2 servers, which
+could be slow for large numbers of instances. This RFD replaces the process on Azure
+as well.
+
+## Discovery
+
+Azure discovery will be handled by the new [discovery service](https://github.com/gravitational/teleport/blob/master/rfd/0057-automatic-aws-server-discovery.md#discovery) described in RFD 57.
+
+```yaml
+discovery_service:
+  enabled: "yes"
+  azure:
+    - types: ["vm"]
+      regions: ["westcentralus"]
+      tags:
+        "teleport": "yes"
+```
+
+The Teleport SSH service will need a [service principal](https://learn.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?view=azure-cli-latest) with a role that includes the `Microsoft.Compute/virtualMachines/read`
+permission to list virtual machines via the Go Azure SDK.
+
+As with AWS database discover and EC2 discover, new Azure nodes will be discovered
+periodically on a 60 second timer, as new nodes are found they will be added to the
```

Comment:
I'm curious how we settled on 60 seconds here. Databases and desktops use a 5m discovery interval, for example.

Reply:
@zmb3 As every hardcoded timeout whatever value we pick there will be always questions why x not y. Is there any reason to not include the timeout in the config so it can be changed and use 5 min as a default?

Reply:
This is not a timeout, but an interval for periodic operations. We mostly just use hard-coded constants for those. 

I don't feel strongly either way, 1m just feels pretty frequent for something like this.